cpp/examples/nested_types/deduplication.cpp

```diff
@@ -0,0 +1,205 @@
+  auto builder   = cudf::io::json_writer_options::builder(sink_info, tbl).lines(true);
+  builder.metadata(metadata);
+  auto options = builder.build();
+  cudf::io::write_json(options);
+}
+
+/**
+ * @brief Aggregate count of duplicate rows in nested-type column
+ */
+std::unique_ptr<cudf::table> count_aggregate(cudf::table_view tbl)
+{
+  // Get count for each key
+  auto keys = cudf::table_view{{tbl.column(0)}};
+  auto val  = cudf::make_numeric_column(cudf::data_type{cudf::type_id::INT32}, keys.num_rows());
+
+  cudf::groupby::groupby grpby_obj(keys);
+  std::vector<cudf::groupby::aggregation_request> requests;
+  requests.emplace_back(cudf::groupby::aggregation_request());
+  auto agg = cudf::make_count_aggregation<cudf::groupby_aggregation>();
+  requests[0].aggregations.push_back(std::move(agg));
+  requests[0].values = *val;
+  auto agg_results   = grpby_obj.aggregate(requests);
+  auto result_key    = std::move(agg_results.first);
+  auto result_val    = std::move(agg_results.second[0].results[0]);
+
+  auto left_cols = result_key->release();
+  left_cols.push_back(std::move(result_val));
+
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+/**
+ * @brief Join each row with its duplicate counts
+ */
+std::unique_ptr<cudf::table> join_count(cudf::table_view left, cudf::table_view right)
+{
+  auto [left_indices, right_indices] =
+    cudf::inner_join(cudf::table_view{{left.column(0)}}, cudf::table_view{{right.column(0)}});
+  auto new_left  = cudf::gather(left, cudf::device_span<int const>{*left_indices});
+  auto new_right = cudf::gather(right, cudf::device_span<int const>{*right_indices});
+
+  auto left_cols  = new_left->release();
+  auto right_cols = new_right->release();
+  left_cols.push_back(std::move(right_cols[1]));
+
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+/**
+ * @brief Sort nested-type column
+ */
+std::unique_ptr<cudf::table> sort_keys(cudf::table_view tbl)
+{
+  auto sort_order = cudf::sorted_order(cudf::table_view{{tbl.column(0)}});
+  return cudf::gather(tbl, *sort_order);
+}
+
+/**
+ * @brief Main for nested_types examples
+ *
+ * Command line parameters:
+ * 1. JSON input file name/path (default: "example.json")
+ * 2. JSON output file name/path (default: "output.json")
+ * 3. Memory resource (optional): "pool" or "cuda" (default: "pool")
+ *
+ * The stdout includes the number of rows in the input and the output size in bytes.
+ */
+int main(int argc, char const** argv)
+{
+  std::string input_filepath;
+  std::string output_filepath;
+  std::string mr_name;
+  if (argc < 4) {
+    input_filepath  = "example.json";
+    output_filepath = "output.json";
+    mr_name         = "pool";
+  } else if (argc == 4) {
+    input_filepath  = argv[1];
+    output_filepath = argv[2];
+    mr_name         = argv[3];
+  } else {
+    std::cout << "Either provide all command-line arguments, or none to use defaults" << std::endl;
+    return 1;
+  }
```

Comment:
suggestion (non-blocking): this error isn't quite doing what you want. It's probably better to be explicit here.

```suggestion
  std::string input_filepath;
  std::string output_filepath;
  std::string mr_name;
  if (argc != 4 && argc != 1) {
    std::cout << "Either provide all command-line arguments, or none to use defaults" << std::endl;
    return 1;
  }
  if (argc == 1) {
    input_filepath  = "example.json";
    output_filepath = "output.json";
    mr_name         = "pool";
  } else {
    input_filepath  = argv[1];
    output_filepath = argv[2];
    mr_name         = argv[3];
  }
```